rust/processor/src/worker.rs

```diff
@@ -1126,23 +1122,20 @@ pub async fn create_fetcher_loop(
             if is_success {
                 continue;
             }
-            if let Some(lrt) = last_reconnection_time {
-                let elapsed: u64 = lrt.elapsed().as_secs();
-                if reconnection_retries >= RECONNECTION_MAX_RETRIES
-                    && elapsed < MIN_SEC_BETWEEN_GRPC_RECONNECTS
-                {
-                    error!(
-                        processor_name = processor_name,
-                        service_type = PROCESSOR_SERVICE_TYPE,
-                        stream_address = indexer_grpc_data_service_address.to_string(),
-                        seconds_since_last_retry = elapsed,
-                        "[Parser] Recently reconnected. Will not retry.",
-                    );
-                    panic!("[Parser] Recently reconnected. Will not retry.")
-                }
+
+            // Sleep for 100ms between reconnect tries
+            tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
I would recommend an exponential retry strategy - ala https://docs.rs/tokio-retry/latest/tokio_retry/ - so that we try reconnecting the first time quickly, to minimize latency, and fall off if we realize the end service might be down, restarting, etc

Reply:
Currently the exponential backoff is managed in the pods and it's obscuring some of the issues we're trying to test/reproduce with indexer GRPC. This PR moves the reconnect logic to be managed within code. I'll add a todo to add exponential backoff.

Reply:
ah, pods didn't come back up immediately?

Reply:
No I've seen it taking > 15s for a pod to restart after crash loop backoff. Removing the backoff fixed the problem we were seeing, so I'll keep this and revisit backoff later.

Reply:
On the infra level there are also resource contention problems. E.g. there are too many pods and we hit a quota or something.